lib/std/array_list.zig

```diff
@@ -108,6 +108,13 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
             return result[0 .. result.len - 1 :sentinel];
         }
 
+        /// Creates a copy of this ArrayList, using the same allocator.
+        pub fn clone(self: *Self) !Self {
+            var cloned = try ArrayList(T).initCapacity(self.allocator, self.capacity);
```

Comment:
This will erase alignment upon cloning. We can avoid that by using `Self`:
```suggestion
            var cloned = try Self.initCapacity(self.allocator, self.capacity);
```